### PR TITLE
Increment methods for major, minor and patch

### PR DESCRIPTION
**Pull request by edupo**
According to specs.

Signed-off-by: Eduardo Lezcano <eduardo.lezcano@be.atlascopco.com>

### CODE REVIEW

**Comment by hanzei** [comment]
This is a great feature. @blang what needs to be done to get this merged?

**Comment by hanzei** [comment]
@edupo Would you mind also adding some decrement methods?

**Comment by blang** [comment]
@edupo and @hanzei, thanks for you patience. LGTM

File: semver.go [hunk 1]
```diff
@@ -161,6 +161,36 @@ func (v Version) Compare(o Version) int {
 
 }
 
+// IncrementPatch increments the patch version
+func (v *Version) IncrementPatch() error {
+	if v.Major == 0 {
+		return fmt.Errorf("Patch version can not be incremented for %q", v.String())
```
Comment (yufeiminds) [inline semver.go]: Why raise an error when major is zero?

**Pull request merged**